src/stores/chat-slice.ts

```diff
@@ -0,0 +1,384 @@
+					if ( error ) {
+						return reject( error );
+					}
+					return resolve( { data, headers } );
+				}
+			);
+		} );
+
+		return {
+			chatApiId: data?.id,
+			maxQuota: headers[ 'x-quota-max' ] || '',
+			message: data?.choices?.[ 0 ]?.message?.content,
+			messageApiId: data?.choices?.[ 0 ]?.message?.id,
+			remainingQuota: headers[ 'x-quota-remaining' ] || '',
+		};
+	}
+);
+
+type SendFeedbackParams = {
+	client: WPCOM;
+	instanceId: string;
+	messageApiId: number;
+	ratingValue: number;
+};
+
+export const sendFeedbackThunk = createAsyncThunk(
+	'chat/sendFeedback',
+	async ( { client, messageApiId, ratingValue, instanceId }: SendFeedbackParams, thunkAPI ) => {
+		const state = thunkAPI.getState() as RootState;
+		const chatApiId = state.chat.chatApiIdDict[ instanceId ];
+
+		try {
+			await client.req.post( {
+				path: `/odie/chat/wpcom-studio-chat/${ chatApiId }/${ messageApiId }/feedback`,
+				apiNamespace: 'wpcom/v2',
+				body: {
+					rating_value: ratingValue,
+				},
+			} );
+		} catch ( error ) {
+			Sentry.captureException( error );
+			console.error( error );
+		}
+	}
+);
+
+const storedMessages = localStorage.getItem( CHAT_MESSAGES_STORE_KEY );
+const storedChatIds = localStorage.getItem( CHAT_ID_STORE_KEY );
+const EMPTY_MESSAGES: readonly Message[] = Object.freeze( [] );
+
+export interface ChatState {
+	currentURL: string;
+	pluginListDict: Record< string, string[] >;
+	themeListDict: Record< string, string[] >;
+	numberOfSites: number;
+	phpVersion: string;
+	siteName: string;
+	themeName: string;
+	isBlockTheme: boolean;
+	os: string;
+	availableEditors: string[];
+	wpVersion: string;
+	messagesDict: { [ key: string ]: Message[] };
+	chatApiIdDict: { [ key: string ]: string | undefined };
+	chatInputBySite: { [ key: string ]: string };
+	isLoadingDict: Record< string, boolean >;
+	promptUsageDict: Record< string, { maxQuota: string; remainingQuota: string } >;
+}
+
+const initialState: ChatState = {
+	currentURL: '',
+	pluginListDict: {},
+	themeListDict: {},
+	numberOfSites: 0,
+	themeName: '',
+	wpVersion: '',
+	phpVersion: DEFAULT_PHP_VERSION,
+	isBlockTheme: false,
+	os: window.appGlobals?.platform || '',
+	availableEditors: [],
+	siteName: '',
+	messagesDict: storedMessages ? JSON.parse( storedMessages ) : {},
+	chatApiIdDict: storedChatIds ? JSON.parse( storedChatIds ) : {},
+	chatInputBySite: {},
+	isLoadingDict: {},
+	promptUsageDict: {},
+};
+
+export function generateMessage(
+	content: string,
+	role: 'user' | 'assistant',
+	newMessageId: number,
+	chatApiId?: string,
+	messageApiId?: number
+): Message {
+	return {
+		content,
+		role,
+		id: newMessageId,
+		chatApiId,
+		createdAt: Date.now(),
+		feedbackReceived: false,
+		messageApiId,
+	};
+}
+
+const chatSlice = createSlice( {
+	name: 'chat',
+	initialState,
+	reducers: {
+		resetChatState: ( state ) => {
+			// Reassigning `initialState` to `state` doesn't work. Probably because of Immer.js.
+			Object.assign( state, initialState );
+		},
```

Comment:
Do you mean `state = initialState`? If so, it's because arguments don't work by reference, as in PHP with prefix `&`. So with assigning you just change the variable inside this function.

Anyay, I found more right approach https://stackoverflow.com/questions/59424523/reset-state-to-initial-with-redux-toolkit

```suggestion
		resetChatState: ( state ) => initialState,
```

Reply:
Returning `initialState` makes sense 👍 Less of a workaround